metadata/metadata.go

```diff
@@ -259,9 +297,11 @@ func (wrapper SearchWrapper) Set(id ResourceID, res Resource) error {
 
 type LocalResourceLookup map[ResourceID]Resource
 
-func (lookup LocalResourceLookup) Lookup(id ResourceID) (Resource, error) {
+func (lookup LocalResourceLookup) Lookup(ctx context.Context, id ResourceID) (Resource, error) {
+	logger := logging.GetLoggerFromContext(ctx)
 	res, has := lookup[id]
 	if !has {
+		logger.Errorw("resource not found", "resource ID", id.String())
```

Comment:
Should add wrapped to this in "err", not a big deal but again include the error when you can in logging